### ISSUE TITLE
[Question] Run tests without exiting the process

### ISSUE THREAD

**Issue by Smrtnyk** [microsoft/playwright#11129]
We wrote some tests in playwright and want to excercise our CI for a few weeks.
Tests would be executed in CI stages on all PR builds, but we don't want to fail the pipeline if playwright tests fail, we just want to observe it for some time to learn about how it behaves, does it fail often etc...
Is there a flag that would not exit the process if tests fail?

**Comment by Smrtnyk** [comment]
Also are there some best practices for CI when it comes to playwright written somewhere?
For example is it ok to run all the tests in parallel in CI?

Jest recommends using `maxWorkers: "50%"` in CI. But not sure for playwright what is recommended.
Our nodes in CI are not always the same, we sometimes get different machine assigned with less or more resources and cores.

**Comment by aslushnikov** [comment]
> Is there a flag that would not exit the process if tests fail?

@Smrtnyk Do you use @playwright/test? In this case, test runner would not exit as a test fails. It will, however, exit with non-zero error code. Depending on your CI, you can instruct it to continue executing other steps, e.g. in github actions you'd use the [`if: always()` condition](https://stackoverflow.com/a/58859404/314883).

> Also are there some best practices for CI when it comes to playwright written somewhere?

We have some documentation available here: https://playwright.dev/docs/ci

> For example is it ok to run all the tests in parallel in CI? Jest recommends using maxWorkers: "50%" in CI. But not sure for playwright what is recommended.

By default, Playwright uses `cpu_count / 2` processes to run tests - this is the same as `maxWorkers: 50%` jest setting. You can read more about parallelization here: https://playwright.dev/docs/test-parallel

> Another thing I have noticed is that sometimes test.skip condition is not respected
> I am using
> test.skip(browserName === "firefox"); inside the test but sometimes the same test times out on firefox when it should be skipped.

Not sure about this. Can you file a separate issue with a test snippet?

**Comment by Smrtnyk** [comment]
Yeah I am using @playwright/test. I have wrapped the step with catchError in jenkins file.

About the timeout issue, what I observed from excercising our CI is that bad guy is mostly the firefox.
Chromium and Webkit are pretty stable.
Firefox occasionally just timeouts and I think the issue happens even before it sees the test.skip statement in the test, therefore it looks like it also timeouts on skipped tests.
Also I sometimes see the same issue as #11130 and it is just on firefox and just happens sometimes.

Our testing case is a bit complex since we test an apm tool and not the UI, so we intercept routes and inject the script in the html before it is served. We use playwright request intercept to serve html, and not some server script.

Will try to make simple test case where I can reproduce it sometimes.
I am running tests on windows locally, but in CI we have linux so seems like it happens regardless of the platform.

**Comment by Smrtnyk** [comment]
> 
> By default, Playwright uses `cpu_count / 2` processes to run tests - this is the same as `maxWorkers: 50%` jest setting. You can read more about parallelization here: https://playwright.dev/docs/test-parallel
> 

In the link you provided which I already read before the example shows
```ts
// Limit the number of workers on CI, use default locally
workers: process.env.CI ? 2 : undefined,
```

Is there are a reason maybe why the example suggests that conditional statement for the CI, or is that just some arbitrary example?

**Comment by Smrtnyk** [comment]
I saw in another issue that there were some bugs fixed for firefox and that will come in next release.
Will close this for now since our CI seems pretty much stable, other than occasional ff hiccups.
